### PR TITLE
Update HiGlass schema

### PR DESCRIPTION
**Pull request by alexander-veit**
This PR updates the schema for Higlass viewconfigs. Many redundant or incorrect fields have been removed from the schema. Only fields that are globally present in every viewconf with the same default value (that are different from the Higlass defaults) and fields that are supposed to be indexed remain. 
I deleted most of the track options from the schema. Default values for track options are now taken care of in visualization.py. 
Track names (labels) and uuids remain in the schema, as we might want to search for those.

### CODE REVIEW

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740&bid=20629688)

File: src/encoded/tests/data/inserts/higlass_view_config.json [hunk 1]
```diff
@@ -49,39 +46,30 @@
...
                                     "color": "#777777",
                                     "labelColor": "#888",
                                     "mousePositionColor": "#999999",
                                     "fontSize": 12,
                                     "labelPosition": "hidden",
                                     "trackBorderWidth": 0,
-                                    "minusStrandColor": "red",
                                     "stroke": "#FFFFFF",
-                                    "showTooltip": false,                                    "showMousePosition": false,
+                                    "showTooltip": false,
```
Comment (j1z0) [inline src/encoded/tests/data/inserts/higlass_view_config.json:65]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/tests/test_higlass.py [hunk 2]
```diff
@@ -2041,15 +1949,15 @@ def test_remove_1d(testapp, higlass_mcool_viewconf, chromsizes_file_json,
...
         if "uid" in track:
             assert_true(
                 track["uid"] not in top_track_uids,
                 "Top track uid reused for left track: {uid}".format(
                     uid=track["uid"]))
         if track["type"] in types_to_find:
             types_to_find[track["type"]] += 1
-        if track["orientation"] == "1d-vertical":
-            vertical_tracks_found += 1
+
```
Comment (j1z0) [inline src/encoded/tests/test_higlass.py:1960]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

**Review by willronchetti — APPROVED** [review]
I would consider deploying a change like this to hotseat and reindexing higlass_view_config to be sure all is good. Also needs a version bump. Other than that LGTM.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 18
           

Clones removed
==============
+ src/encoded/static/components/item-pages/HiGlassViewConfigView.js  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740&bid=20629688)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 3]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+      // Gets all tracks with the same orientaton (e.g. all top tracks)
+      const tracks = currentViewConf.views[trackInfo.vIndex].tracks[trackInfo.track];
+      const currentTrack = _.find(tracks, function (t) {
+        return t.uid === trackInfo.uid;
+      });
+
+      if (currentTrack) {
+        const similarTilesets = _.filter(
+          tracks,
+          (trackItem) => currentTrack.type === trackItem.type
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:385]: Codacy found an issue: [Expected indentation of 16 spaces but found 10.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 4]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
-        }
-        return false;
+      const hgc = this.getHiGlassComponent();
+      const currentViewConf = this.getHiGlassViewConfig(hgc);
+
+      if (!currentViewConf) {
+        throw new Error("Could not get current view configuration.");
+      }
+
+      // Gets all tracks with the same orientaton (e.g. all top tracks)
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:376]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 5]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
-            const similarTilesets = _.filter(track, (tilesetItem) => orientation === tilesetItem.orientation && type === tilesetItem.type);
-            return similarTilesets && similarTilesets.length > 1;
-        }
-        return false;
+      const hgc = this.getHiGlassComponent();
+      const currentViewConf = this.getHiGlassViewConfig(hgc);
+
+      if (!currentViewConf) {
+        throw new Error("Could not get current view configuration.");
+      }
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:374]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 6]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+
+      // Gets all tracks with the same orientaton (e.g. all top tracks)
+      const tracks = currentViewConf.views[trackInfo.vIndex].tracks[trackInfo.track];
+      const currentTrack = _.find(tracks, function (t) {
+        return t.uid === trackInfo.uid;
+      });
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:380]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 7]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+      const currentViewConf = this.getHiGlassViewConfig(hgc);
+
+      if (!currentViewConf) {
+        throw new Error("Could not get current view configuration.");
+      }
+
+      // Gets all tracks with the same orientaton (e.g. all top tracks)
+      const tracks = currentViewConf.views[trackInfo.vIndex].tracks[trackInfo.track];
+      const currentTrack = _.find(tracks, function (t) {
+        return t.uid === trackInfo.uid;
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:379]: Codacy found an issue: [Expected indentation of 12 spaces but found 8.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 8]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
-        return false;
+      const hgc = this.getHiGlassComponent();
+      const currentViewConf = this.getHiGlassViewConfig(hgc);
+
+      if (!currentViewConf) {
+        throw new Error("Could not get current view configuration.");
+      }
+
+      // Gets all tracks with the same orientaton (e.g. all top tracks)
+      const tracks = currentViewConf.views[trackInfo.vIndex].tracks[trackInfo.track];
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:377]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 9]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+      const currentTrack = _.find(tracks, function (t) {
+        return t.uid === trackInfo.uid;
+      });
+
+      if (currentTrack) {
+        const similarTilesets = _.filter(
+          tracks,
+          (trackItem) => currentTrack.type === trackItem.type
+        );
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:386]: Codacy found an issue: [Expected indentation of 16 spaces but found 20. (indent)](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 10]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+        return similarTilesets && similarTilesets.length > 1;
+      }
+      return false;
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:389]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 11]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+      const currentTrack = _.find(tracks, function (t) {
+        return t.uid === trackInfo.uid;
+      });
+
+      if (currentTrack) {
+        const similarTilesets = _.filter(
+          tracks,
+          (trackItem) => currentTrack.type === trackItem.type
+        );
+        return similarTilesets && similarTilesets.length > 1;
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:387]: Codacy found an issue: [Expected indentation of 12 spaces but found 8.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 12]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
-        const track = currentViewConf.views[trackInfo.vIndex].tracks[trackInfo.track];
-        const tileset = _.find(track, function (t) { return t.uid === trackInfo.uid; });
-
-        if (tileset) {
-            const { orientation, type } = tileset;
-            const similarTilesets = _.filter(track, (tilesetItem) => orientation === tilesetItem.orientation && type === tilesetItem.type);
-            return similarTilesets && similarTilesets.length > 1;
-        }
-        return false;
+      const hgc = this.getHiGlassComponent();
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:369]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 13]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+      const currentViewConf = this.getHiGlassViewConfig(hgc);
+
+      if (!currentViewConf) {
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:372]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 14]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+        throw new Error("Could not get current view configuration.");
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:373]: Codacy found an issue: [Expected indentation of 12 spaces but found 8.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/tests/test_higlass.py [hunk 15]
```diff
@@ -2041,16 +1949,15 @@ def test_remove_1d(testapp, higlass_mcool_viewconf, chromsizes_file_json,
...
             assert_true(
                 track["uid"] not in top_track_uids,
                 "Top track uid reused for left track: {uid}".format(
                     uid=track["uid"]))
         if track["type"] in types_to_find:
             types_to_find[track["type"]] += 1
-        if track["orientation"] == "1d-vertical":
-            vertical_tracks_found += 1
-
+
```
Comment (j1z0) [inline src/encoded/tests/test_higlass.py:1960]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 16]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+      }
+
+      // Gets all tracks with the same orientaton (e.g. all top tracks)
+      const tracks = currentViewConf.views[trackInfo.vIndex].tracks[trackInfo.track];
+      const currentTrack = _.find(tracks, function (t) {
+        return t.uid === trackInfo.uid;
+      });
+
+      if (currentTrack) {
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:382]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 17]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+        const similarTilesets = _.filter(
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:383]: Codacy found an issue: [Expected indentation of 12 spaces but found 8.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 18]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
-        const tileset = _.find(track, function (t) { return t.uid === trackInfo.uid; });
-
-        if (tileset) {
-            const { orientation, type } = tileset;
-            const similarTilesets = _.filter(track, (tilesetItem) => orientation === tilesetItem.orientation && type === tilesetItem.type);
-            return similarTilesets && similarTilesets.length > 1;
-        }
-        return false;
+      const hgc = this.getHiGlassComponent();
+      const currentViewConf = this.getHiGlassViewConfig(hgc);
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:370]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 19]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+
+      // Gets all tracks with the same orientaton (e.g. all top tracks)
+      const tracks = currentViewConf.views[trackInfo.vIndex].tracks[trackInfo.track];
+      const currentTrack = _.find(tracks, function (t) {
+        return t.uid === trackInfo.uid;
+      });
+
+      if (currentTrack) {
+        const similarTilesets = _.filter(
+          tracks,
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:384]: Codacy found an issue: [Expected indentation of 16 spaces but found 10.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

File: src/encoded/static/components/item-pages/HiGlassViewConfigView.js [hunk 20]
```diff
@@ -359,26 +361,32 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
...
+      const hgc = this.getHiGlassComponent();
+      const currentViewConf = this.getHiGlassViewConfig(hgc);
+
+      if (!currentViewConf) {
+        throw new Error("Could not get current view configuration.");
+      }
+
+      // Gets all tracks with the same orientaton (e.g. all top tracks)
+      const tracks = currentViewConf.views[trackInfo.vIndex].tracks[trackInfo.track];
+      const currentTrack = _.find(tracks, function (t) {
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/HiGlassViewConfigView.js:378]: Codacy found an issue: [Expected indentation of 8 spaces but found 6.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

**Review by utku-ozturk — APPROVED** [review]
I have tested locally, works great! I only pushed a minor commit to fix the indentation/formatting warning.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones removed
==============
+ src/encoded/static/components/item-pages/HiGlassViewConfigView.js  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740&bid=20629688)

File: src/encoded/tests/test_higlass.py (hunk 15 above)
Comment (j1z0) [inline src/encoded/tests/test_higlass.py:1960]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones removed
==============
+ src/encoded/static/components/item-pages/HiGlassViewConfigView.js  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740&bid=20629688)

File: src/encoded/tests/test_higlass.py (hunk 15 above)
Comment (j1z0) [inline src/encoded/tests/test_higlass.py:1960]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=6353740)

**Pull request merged**